_topics/en/transaction-origin-privacy.md

```diff
@@ -0,0 +1,106 @@
+
+  - title: "Transcript: TxProbe: discovering Bitcoin's network topology using orphan transactions"
+    url: /en/newsletters/2019/09/18/#txprobe-discovering-bitcoin-s-network-topology-using-orphan-transactions
+
+  - title: "Bitcoin Core #14897 introduces a semi-random order when requesting transactions"
+    url: /en/newsletters/2019/02/12/#bitcoin-core-14897
+
+  - title: "Bitcoin Core #19109 adds a per-peer rolling bloom filter to track recent tx announcements"
+    url: /en/newsletters/2020/07/22/#bitcoin-core-19109
+
+## Optional.  Same format as "primary_sources" above
+see_also:
+  - title: Anonymity networks
+    link: topic anonymity networks
+
+  - title: V2 P2P Transport
+    link: topic v2 p2p transport
+
+  - title: Dandelion
+    link: topic dandelion
+
+## Optional.  Force the display (true) or non-display (false) of stub
+## topic notice.  Default is to display if the page.content is below a
+## threshold word count
+#stub: false
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **Transaction origin privacy** is the ability to hide the origin of a
+  transaction from network surveillance.
+---
+If an eavesdropper can see every transaction relayed by every node, they
+can assume that the first node to send a transaction is the node where
+that transaction originated.  This implies that someone at that node's
+IP address created that transaction.  Whether that implication is
+correct or not, this capability is harmful to user privacy.
+
+Even when surveillance of the entire Bitcoin network is not possible,
+there are many methods for attempting to locate which node originated a
+transaction.  Developers have expended significant effort into improving
+transaction origin privacy.  Some previous and ongoing efforts include:
+
+- **Lightning Network:** although onchain LN transactions still need
+  transaction origin privacy, offchain LN payments which are routed
+  through onion-encrypted messages are more resistant to many of the
+  techniques applied against onchain Bitcoin transactions.
+
+- [Anonymity networks][topic anonymity networks]: support has been added
+  to Bitcoin Core for encrypted relay of transaction and other data over
+  Tor, I2P, and CJDNS.  Several other Bitcoin programs support some or
+  all of these protocols.  Encryption makes it much more difficult for
+  an evesdropper to reliably determine the origin of a transaction.
+
+- [v2 P2P transport][topic v2 p2p transport]: this protocol will provide
+  native encryption of transaction and other data for Bitcoin programs.
+
+- [Dandelion][topic dandelion]: this protocol will allow a node which
+  did not originate a transaction to be the first to publicly broadcast
+  it, potentially making it much more difficult to determine the actual
+  origin.  Dandelion will be much more effective if it is combined with
+  encryption, either naively or through anonymity networks.
```

Comment:
I thought that Dandelion has basically been given up on, since the DoS vectors it introduces could not be efficiently mitigated. From what I understand, all the concerns described by sdaftuar in this [BSE post](https://bitcoin.stackexchange.com/a/81504/5406) still apply

Reply:
There was discussion in bitcoin-core-dev IRC recently about "[one-hop dandelion](https://www.erisian.com.au/bitcoin-core-dev/log-2023-04-22.html#l-76)", which (I just learned) means having the origination node announce the transaction over an anonymity network (e.g. Tor) but not store it in its own mempool until it receives it from the P2P network.  In other words, to announce the transaction using an ephemeral identity and then act like any other node when it later sees that transaction.

Should that properly be called "dandelion"?  I dunno.  But it seems like dandelion is still a concept that several developers are thinking about, so I think it's beneficial to list it as something for new learners.